viadot/task_utils.py

```diff
@@ -527,3 +530,41 @@ def git_clone_url(self):
         if self.use_ssh:
             return f"git@{self.repo_host}:{self.repo}"
         return f"https://{self.git_token_secret}@{self.repo_host}/{self.repo}"
+
+
+@task
+def credentials_loader(credentials_secret: str, vault_name: str = None) -> dict:
+    """
+    Function that gets credentials from azure Key Vault or PrefectSecret or from prefect local config.
+    Args:
+        credentials_secret (str): The name of the Azure Key Vault secret containing a dictionary
+        with credentials.
+        vault_name (str, optional): The name of the vault from which to obtain the secret. Defaults to None.
+    Returns: Credentials
+    """
+    from viadot.tasks.azure_key_vault import AzureKeyVaultSecret
+
+    if credentials_secret:
+        try:
+            credentials = local_config[credentials_secret]
+            logger.info("Successfully loaded credentials from local config.")
+        except (ValueError, KeyError):
+            try:
+                azure_secret_task = AzureKeyVaultSecret()
+                credentials_str = azure_secret_task.run(
+                    secret=credentials_secret, vault_name=vault_name
+                )
+                credentials = json.loads(credentials_str)
+                logger.info("Successfully loaded credentials from Azure Key Vault.")
+            except Exception:
+                try:
+                    credentials = PrefectSecret(credentials_secret).run()
```

Comment:
Maybe will be a good idea to add KeyVault support in the `PrefectSecret` section (in this file are 2 options: 1) key, 2) credentials).